src/utility.h

```diff
@@ -224,6 +224,17 @@ class FileStatus :
         ::stat{}
     {}
 
+    FileStatus(const FileStatus &other)
+    {
+        *this = other;
+    }
+
+    FileStatus& operator=(const FileStatus &other)
+    {
+        std::memcpy(this, &other, sizeof(struct stat));
```

Comment:
Originally I intended this `FileStatus` class to be only a thing wrapper around the `struct stat`. Since coupling this notorious C data structure with C++ semantics is quite peculiar and hard to get right or feeling natural. That was also why I had that explicit `copy()` operation in the first place. So in hindsight making `struct stat` a member of `FileStatus` would have been cleaner but less conventient.

Let's follow your suggestion then to make it use the C++ notion all round.